fcrepo-persistence-api/src/main/java/org/fcrepo/persistence/api/CreateResourceRequest.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.api;
+
+/**
+ * A class that holds data required for creating a resource.
+ * @author dbernstein
+ */
+public class CreateResourceRequest {
```

Comment:
Seems like the `CreateResourceRequest`, `DeleteResourceRequest` and `UpdateResourceRequest` could be a very thin wrapper around a base class as they will probably all need the information from the initial HTTP request. 

Which then leads me to ask if perhaps 3 separate classes is unnecessary and a `getOperation()` which could return `CREATE`, `UPDATE` or `DELETE` is sufficient. This class doesn't do any work it just shuttles information around, right?

Reply:
Right - I just thought it would be clearer to have a clearly defined set of parameters that go with each operation.  This PR is out of date, and Ben's latest changes supersede it but I think it still makes sense to have separate classes for each  operation (as is reflected in Ben's PR)